htdocs/societe/card.php

```diff
@@ -2936,7 +2939,43 @@ function init_supplier_categ() {
 		include DOL_DOCUMENT_ROOT.'/core/tpl/card_presend.tpl.php';
 	}
 }
-
+/**
+ * @param string $valuetoprint the value to print
+ * @param string $spanid unique id for a span
+ * @param Translate $langs the $langs variable
+ * @return string the string to print for the button
+ */
+function showValueWithCopyAndPasteButton($valuetoprint, $spanid, $langs)
+{
+	$stringtoprint= '<span id="'.$spanid.'">';
+	$stringtoprint.= $valuetoprint .'</span>&ensp;';
+	$stringtoprint.= '</span>';
+	$stringtoprint.='<span class="opacitymedium"  id="'.dol_escape_json($spanid).'clipboardChar" onclick="'.dol_escape_json($spanid).'clipBoardCopy()"><i " class="far fa-clipboard"></i></span>';
+	$stringtoprint.="<script>
+		function ".dol_escape_json($spanid)."clipBoardCopy() {
+			var copied_text = document.getElementById( \"".dol_escape_json($spanid)."\" );
+			if ( window.getSelection ) {
+				
+				selection = window.getSelection();
+				
+				range = document.createRange();
+				range.selectNodeContents( copied_text );
+				
+				selection.removeAllRanges();
+				selection.addRange( range );
+				
+			}
+			
+			document.execCommand( 'copy' );
+			window.getSelection().removeAllRanges();
+			var span = document.getElementById( \"".dol_escape_json($spanid)."clipboardChar\" );
+			var tmp = span.innerHTML
+			span.innerHTML = '".$langs->trans('CopiedToClipboard')."';
```

Comment:
Because a translation can be overwrite, to avoid js injection, please add dol_escapejs() around the result of translation.